predicators/main.py

```diff
@@ -93,13 +97,13 @@ def main() -> None:
     preds, _ = utils.parse_config_excluded_predicates(env)
     # Create the train tasks.
     env_train_tasks = env.get_train_tasks()
-    # This is a placeholder for a forthcoming perception API. However, we will
-    # make the assumption for the foreseeable future that a train Task can be
-    # constructed from a train EnvironmentTask. In other words, the initial obs
-    # is assumed to contain enough information to determine all of the objects
-    # and their initial states. We only make this assumption for the training
-    # tasks, we don't need to make it for the test tasks.
-    train_tasks = [t.task for t in env_train_tasks]
+    # We assume that a train Task can be constructed from a EnvironmentTask.
+    # In other words, the initial obs is assumed to contain enough information
+    # to determine all of the objects and their initial states. We only make
+    # this assumption for the training tasks, we don't need to make it for the
+    # test tasks. It is also something we can weaken later if needed.
+    perceiver = create_perceiver(CFG.perceiver)
+    train_tasks = [perceiver.reset(t) for t in env_train_tasks]
```

Comment:
Hold on - I'm pretty confused now, because this is exactly what I would've expected! If we're using the perceiver to generate train tasks for us, then why are we directly getting init state and goal from the `EnvironmentTask` in `base_env.py`. Don't we want to be using the perceiver there as well in exactly this way?

Reply:
well we certainly don't want to be using the perceiver inside of the environment because that would break the environment/agent boundary, but you're right that using `EnvironmentTask.task` inside of the environment to check if the goal holds is not ideal. I'm only doing it out of convenience because all of the existing environment (except sokoban) have goal descriptions that are actually goals (ground atoms). but maybe a step in the right direction would be to change this implementation to assert that the environment task goal description is a set of ground atoms before checking the goal? and then I can also add a comment to clarify things?

Reply:
I think that would help! iirc, you already check that somewhere, but explicitly checking it in `base_env.py` I think would help clarify things